packages/shared/src/globalsAllowList.ts

```diff
@@ -1,9 +1,41 @@
 import { makeMap } from './makeMap'
 
-const GLOBALS_ALLOWED =
-  'Infinity,undefined,NaN,isFinite,isNaN,parseFloat,parseInt,decodeURI,' +
-  'decodeURIComponent,encodeURI,encodeURIComponent,Math,Number,Date,Array,' +
-  'Object,Boolean,String,RegExp,Map,Set,JSON,Intl,BigInt,console'
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects
+const GLOBAL_SIMPLE_VALUE = 'globalThis,Infinity,NaN,undefined'
+const FUNCTION =
+  'eval,isFinite,isNaN,parseFloat,parseInt,decodeURI,decodeURIComponent,encodeURI,encodeURIComponent'
+const FUNDAMENTAL = 'Object,Function,Boolean,Symbol'
+const ERROR =
+  'Error,AggregateError,EvalError,RangeError,ReferenceError,SyntaxError,TypeError,URIError'
+const NUMBER_AND_DATES = 'Math,BigInt,Number,Date'
+const TEXT = 'String,RegExp'
+const INDEXED_COLLECTIONS =
+  'Array,Int8Array,Uint8Array,Uint8ClampedArray,Int16Array,Uint16Array,Int32Array,Uint32Array,BigInt64Array,BigUint64Array,Float32Array,Float64Array'
+const KEYED_COLLECTIONS = 'Map,Set,WeakMap,WeakSet'
+const STRUCTURED_DATA = 'ArrayBuffer,SharedArrayBuffer,DataView,Atomics,JSON'
+const MANAGING_MEMORY = 'WeakRef,FinalizationRegistry'
+const CONTROL_ABSTRACTION_OBJECTS =
+  'Iterator,AsyncIterator,Promise,GeneratorFunction,AsyncGeneratorFunction,Generator,AsyncGenerator,AsyncFunction'
+const REFLECTION = 'Reflect,Proxy'
+const INTERNATIONALIZATION = 'Intl'
+const OTHER_GLOBALS_ALLOWED = 'console'
```

Comment:
After discussion, we think `Symbol`, and `Error` can be included, but for others, we can provide a way that can be customized. See https://github.com/vuejs/core/pull/3398/files.